.github/workflows/build-images.yml

```diff
@@ -0,0 +1,48 @@
+
+name: keg-build
+on:
+  push:
+    branches:
+      - ZEN-543-rebuild-workflow
+      - develop
+
+jobs:
+  build-images:
+    name: 'Build Images'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    env:
+      CLI_BRANCH: ZEN-543-updates-for-workflow
+      CLI_PATH: $GITHUB_WORKSPACE/keg-cli
+      CLI_USER: keg-admin
+      GIT_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    steps:
+    - name: "Clone Keg-CLI"
+      id: clone-cli
+      run: git -C $GITHUB_WORKSPACE clone --single-branch --branch ${{env.CLI_BRANCH}} https://github.com/simpleviewinc/keg-cli.git
+    - name: "Install Keg-CLI"
+      id: install-cli
+      run: yarn --cwd $GITHUB_WORKSPACE/keg-cli install
+    - name: "Setup Keg-CLI"
+      id: setup-cli
+      run: cd $GITHUB_WORKSPACE/keg-cli; KEG_HUB_BRANCH=${GITHUB_REF##*/} KEG_ROOT_DIR=$GITHUB_WORKSPACE/keg-hub bash scripts/ci/setupCLIConfig.sh
+    - name: "Set Docker User"
+      id: set-docker-user
+      run: source $GITHUB_WORKSPACE/keg-cli/keg && keg config set key=docker.user value=${{env.CLI_USER}} --confirm false
```

Comment:
Given that each step is inside the same job
  * Adding `source $GITHUB_WORKSPACE/keg-cli/keg` to each step should not be needed. 
  * I have not tested it, but each set should use the same environment, and therefore the file should already be loaded
  * You should just be able to call it once, and be good for all other steps

Another option, would be
  * Call the node script directly with => `node $GITHUB_WORKSPACE/keg-cli/keg-cli <params>`
  * For example, `line 33` would look like this =>
  `run: node $GITHUB_WORKSPACE/keg-cli/keg-cli config set key=docker.user value=${{env.CLI_USER}} --confirm false`

Reply:
Yeah I cleaned this up in the latest commit through refactoring.

But regarding the first option you presented, I actually tried that first but it did not work. [The docs elaborate](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idsteps): 
>Each step runs in its own process in the runner environment and has access to the workspace and filesystem. Because steps run in their own process, changes to environment variables are not preserved between steps. 

Regarding the second option, yeah that would be better (although I don't think it's all that different than just sourcing first), but should no longer be necessary.